### DISCUSSION TITLE
Why is SVD not converging?

### DISCUSSION THREAD

**Discussion by mexicantexan** [colour-science/colour-checker-detection#10] [Q&A]
I was having trouble getting the SVD to converge while using the colour_correction_matrix_Cheung2004 function with the default 3 terms. Ended up figuring out that the array M_T  had ??too small of numbers in it?? My current workaround is to multiply both M_T and M_R by an order of magnitude, which allows the SVD to converge. 

The results seem to be correct, didn't know if this was bad practice?

**Comment by KelSolaar** [comment]
Hi @mexicantexan,

Would it be possible to get the values of the test matrix that is fed to the `colour.characterisation.colour_correction_matrix_Cheung2004` definition? With 3 terms it is simple linear fitting, if that breaks the test values might have some issues.

Cheers,

Thomas

**Comment by mexicantexan** [comment]
Of course @KelSolaar ! Here is the 24x3 array of values that are coming out of the section of test code y'all shared for the X-Rite 24 patch:

[[  9.97202281e-02   3.85103423e-02   1.97636268e-02]
 [  4.62537355e-01   2.42857597e-01   1.63972995e-01]
 [  9.98925304e-02   1.85736228e-01   2.93222063e-01]
 [  6.13056952e-02   9.90630910e-02   2.17022592e-02]
 [  2.02228250e-01   1.96493036e-01   3.63095816e-01]
 [  2.00938278e-01   5.16740398e-01   4.00732241e-01]
 [  6.42805583e-01   1.37695498e-01   6.57558699e-03]
 [  5.52603014e-02   9.39088758e-02   3.65819753e-01]
 [  4.97599167e-01   4.62955672e-02   6.23085985e-02]
 [  5.52022653e-02   1.95881498e-02   7.62875710e-02]
 [  3.53467178e-01   4.86343325e-01   1.88595841e-02]
 [  6.64554393e-01   2.95195579e-01   8.31142248e-04]
 [  1.93091265e-02   3.70667956e-02   2.42407655e-01]
 [  7.44699195e-02   3.18164211e-01   4.63075144e-02]
 [  3.75873531e-01   1.09942182e-02   1.53877934e-02]
 [  7.11079034e-01   5.20607897e-01   1.89704365e-05]
 [  4.50439533e-01   6.34227063e-02   2.21623399e-01]
 [  3.69241240e-02   2.81479470e-01   3.99655835e-01]
 [  7.52550560e-01   7.54553710e-01   6.70833294e-01]
 [  5.38473569e-01   5.44224457e-01   5.01601112e-01]
 [  3.41641462e-01   3.48857309e-01   3.16767197e-01]
 [  1.52460478e-01   1.55573646e-01   1.33169493e-01]
 [  4.98210753e-02   5.13859812e-02   4.38415860e-02]
 [  1.07736835e-02   1.07736835e-02   9.54771158e-03]]

for reference here is the snippet of code that gathers the array above before comparing it to the reference values:
<pre><code> for swatches, colour_checker, masks in detect_colour_checkers_segmentation(COLOUR_CHECKER_IMAGE, additional_data=True):
      self.SWATCHES.append(swatches)
      masks_i = np.zeros(colour_checker.shape)
      for i, mask in enumerate(masks):
          masks_i[mask[0]:mask[1], mask[2]:mask[3], ...] = 1
      if show_output:
          figure, axes, plt1 = plot_image(
              colour.cctf_encoding(
                  np.clip(colour_checker + masks_i * 0.25, 0, 1)), interpolation='lanczos')
          image = self.buffer_plot_and_get(figure, PIL=True)
          image.save('identified_color_card.png')
</code></pre>

**Comment by mexicantexan** [comment]
Oh and here is the test image I am using:  
https://d27vruithpdhv2.cloudfront.net/1xv4ikuo1w/products/68xq31pscv/2D/original/x-rite+colorchecker+copy.jpg

**Comment by KelSolaar** [comment]
Sorry for not coming back to you earlier, it is quite surprising that the fitting fails, the colour looks fine here:

![image](https://user-images.githubusercontent.com/99779/98096385-907c9480-1ef0-11eb-97ee-87dc70a40454.png)

I managed to fit it against this test data: https://github.com/colour-science/colour/blob/develop/colour/characterisation/correction.py#L785 

```python
array([[ 0.74237056, -0.00493562,  0.17259327],
       [ 0.17577396,  0.56242156,  0.19284019],
       [ 0.04939617,  0.01500463,  1.03979242]])
```
